3rdparty/libtengine/tengine.cmake

```diff
@@ -69,23 +69,21 @@ if(BUILD_TENGINE)
 	   elseif(${ANDROID_ABI} STREQUAL "arm64-v8a")
 			   SET(CONFIG_ARCH_ARM64 ON)
 	   endif()
-	   SET(Tengine_LIB "tengine" CACHE INTERNAL "")
 	else()
 		# linux system
 		if(CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
 			   SET(CONFIG_ARCH_ARM32 ON)
 		elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64) ## AARCH64
 			   SET(CONFIG_ARCH_ARM64 ON)
 		endif()
-		SET(Tengine_LIB "tengine" CACHE INTERNAL "")
 	endif()
 
 	SET(BUILT_IN_OPENCV ON) ## set for tengine compile discern .
 	SET(Tengine_INCLUDE_DIR  "${OCV_TENGINE_SOURCE_PATH}/core/include" CACHE INTERNAL "")
+	SET(Tengine_LIB "${CMAKE_BINARY_DIR}/lib/${ANDROID_ABI}/libtengine.a" CACHE INTERNAL "")
```

Comment:
The solution is not good enough too. There are 2 issues:
- TEngine library is not installed. It means that it's missing from Android SDK and other builds that are done with cross-compiler.
- There is build path leak to final CMake config:
```
~/Projects/OpenCV/opencv-master-arm64/install/lib/cmake/opencv4$ grep -Rin tengine
OpenCVModules.cmake:146:  INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:opencv_core>;\$<LINK_ONLY:opencv_imgproc>;opencv_core;opencv_imgproc;\$<LINK_ONLY:dl>;\$<LINK_ONLY:m>;\$<LINK_ONLY:pthread>;\$<LINK_ONLY:rt>;\$<LINK_ONLY:tegra_hal>;\$<LINK_ONLY:libprotobuf>;/home/alexander/Projects/OpenCV/opencv-master-arm64/lib//libtengine.a"
```